test/ProductConstructionService.ScenarioTests/ScenarioTests/ScenarioTests_CodeFlowConflicts.cs

```
@@ -396,7 +395,14 @@ await CreateTargetBranchAndExecuteTest(targetBranchName, testRepoFolder.Director
 
                         // WaitForPullRequestAsync fetches prs in bulk, which doesn't fetch fields like Mergeable and MergeableState which we need
                         pr = await GitHubApi.PullRequest.Get(TestParameters.GitHubTestOrg, TestRepository.TestRepo1Name, pr.Number);
-                        PullRequestShouldHaveConflicts(pr);
+                        try
+                        {
+                            await GitHubApi.Git.Reference.Delete(TestParameters.GitHubTestOrg, TestRepository.TestRepo1Name, $"heads/{pr.Head.Ref}");
+                        }
+                        finally
+                        {
+                            PullRequestShouldHaveConflicts(pr);
```

Why does this need to be in a finally block, can't it just be called before the branch deletion?

---

We make assertions inside which throw but we want to clean up always

---

But we also want to assert even when clean up fails

---

I think we can't extend the try block if we don't have the branch name

---

But it will also always assert if you put the assertions before the branch cleanup. 
I guess that functionally, try { A() } finally { B() } is almost equivalenet to try { B() } finally { A()}
The only difference is that if **both** of them throw an exception, then the first one will be swallowed, and only the second exception will propagate to the test failure

---

I rewrote it to `using`